File: Python/school/inputdigits.py
from quicksort import QuickSort
import sys

class InputDigits:

    def __init__(self):
        self.qs = QuickSort()


    def sort_array(self, array):
        return self.qs.quick_sort(array)

    @staticmethod
    def format_array(array):

        if ',' in str(array):
            return (str(array).replace(',', '')).strip('[]')
        else:
            return str(array).strip('[]')

def main():
    id = InputDigits()

    _input = input('Please enter zero or more digits (or q to quit):')

    if _input == '':
        print('Here are the digits in non-decreasing order: ')

    elif _input.lower() == 'q':
        print('Have a nice day!')
        sys.exit()

    else:
        array = [int(x) for x in _input.split(' ')]
        print('Here are the digits in non-decreasing order: {}'.format(id.format_array(id.sort_array(array))))

if __name__ == '__main__':
    main()

File: Java/FirstJavaProgram/src/InputDigits.java
/**
 * Write a class called InputDigits.

 The program should prompt the user to enter a string consisting
 of zero or more digits in any order with duplicates allowed and
 separated by spaces – be sure to read this in as a single string.
 The program should construct a string with the digits in
 non-decreasing order separated by spaces.
 Finally the program should output the constructed string.

 !!!!Helper Methods are located in the class ProjectUtils!!!!

 */

import java.util.Scanner;

public class InputDigits
{
    private static Scanner in = new Scanner(System.in);

    /**
     * I got sick of typing this shit
     * Really I did, I'm so used to this in python
     * print(); is so much better than System.out.print(); right?
     */
    private static void print(String text)
    {
        System.out.print(text);
    }

    private static void println(String text)
    {
        System.out.println(text);
    }

    public static void main(String[] args)
    {
        println("Please enter zero or more digits (or q to quit):");
        String input = in.nextLine();

        // If they typed q, exit
        if(input.contains("q"))
        {
            ProjectUtils.exit();
        }
        else
        {
            // create the array
            int[] array;

            // Create an empty string
            String arrayString = "";

            // As long as the input isn't 0
            if(input.length() > 0)
            {
                // Make the array an int array
                // sort it, then convert it to a string
                array = ProjectUtils.sortArray(ProjectUtils.makeIntegerArray(input));
                arrayString = ProjectUtils.makeStringArray(array);
            }

            // Print the array, or nothing it the input was nothing
            print("Here are the digits in non-decreasing order: " + arrayString);
        }
    }
}
